Mathlib/FieldTheory/Finite/Basic.lean

```diff
@@ -633,6 +633,42 @@ theorem Int.ModEq.pow_card_sub_one_eq_one {p : ℕ} (hp : Nat.Prime p) {n : ℤ}
     · exact hpn.symm
   simpa [← ZMod.intCast_eq_intCast_iff] using ZMod.pow_card_sub_one_eq_one this
 
+theorem Int.prime_dvd_pow_sub_one {p : ℕ} (hp : Nat.Prime p) {n : ℤ} (hpn : IsCoprime n p) :
+    (p : ℤ) ∣ n ^ (p - 1) - 1 :=
+  modEq_iff_dvd.mp (ModEq.symm (ModEq.pow_card_sub_one_eq_one hp hpn))
+
+theorem Int.prime_dvd_pow_self_sub {p : ℕ} (hp : Nat.Prime p) {n : ℤ} : (p : ℤ) ∣ n ^ p - n := by
+  by_cases ch : (p : ℤ) ∣ n
+  · exact Int.dvd_trans ch (Int.dvd_sub (dvd_pow_self n (Nat.Prime.ne_zero hp)) (Int.dvd_refl n))
+  · have hnp : IsCoprime n p := by
+      rw [ofNat_dvd_left, ← Nat.Prime.coprime_iff_not_dvd hp] at ch
+      exact IsCoprime.symm (isCoprime_iff_gcd_eq_one.mpr ch)
+    refine Int.dvd_trans (prime_dvd_pow_sub_one hp hnp) ⟨n, ?_⟩
+    simp [Int.sub_mul, ← Int.pow_succ, Nat.sub_one_add_one_eq_of_pos (Nat.Prime.one_le hp)]
+
+theorem Int.ModEq.pow_card_eq_self {p : ℕ} (hp : Nat.Prime p) {n : ℤ} : n ^ p ≡ n [ZMOD p] :=
```

Comment:
I don't think `card` should appear in the name (and likewise for `Int.ModEq.pow_card_sub_one_eq_one`).